src/main/java/use_case/leaderboard_as_chart/LeaderboardAsChartInteractor.java

```diff
@@ -0,0 +1,33 @@
+package use_case.leaderboard_as_chart;
+
+import entity.User;
+import frameworks_and_drivers.DataAccess.LeaderboardUserDataAccessObject;
+import use_case.DataAccessException;
+import use_case.leaderboard.LeaderboardType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LeaderboardAsChartInteractor implements LeaderboardAsChartInputBoundary {
+
+    private final LeaderboardUserDataAccessObject dao;
+    private final LeaderboardAsChartOutputBoundary presenter;
+
+    public LeaderboardAsChartInteractor(LeaderboardUserDataAccessObject dao, LeaderboardAsChartOutputBoundary presenter) {
+        this.dao = dao;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void execute() throws DataAccessException {
```

Comment:
```suggestion
    public void execute(LeaderboardAsChartInputData inputData) throws DataAccessException {
```

It is currently **not** passing any inputData sent from the controller. It will work because currently it is not depending on any of the arguments passed by `LeaderboardAsChartController`, but it invalidates the purpose of Clean Architecture.